x-pack/plugins/secops/public/components/timeline/body/renderers/source_dest_ip.tsx

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { getOr } from 'lodash/fp';
+import * as React from 'react';
+import { pure } from 'recompose';
+import styled from 'styled-components';
+
+import { Ecs } from '../../../../graphql/types';
+import {
+  fieldExists,
+  getAllFieldsInSchemaByMappedName,
+  getMappedEcsValue,
+  mappedEcsSchemaFieldNames,
+  virtualEcsSchema,
+} from '../../../../lib/ecs';
+import { escapeQueryValue } from '../../../../lib/keury';
+import { DragEffects, DraggableWrapper } from '../../../drag_and_drop/draggable_wrapper';
+import { escapeDataProviderId } from '../../../drag_and_drop/helpers';
+import { Provider } from '../../data_providers/provider';
+import { FormattedField } from './formatted_field';
+import * as i18n from './translations';
+
+const Label = styled.div`
+  font-weight: bold;
+`;
+
+const allFieldsInSchemaByName = getAllFieldsInSchemaByMappedName(virtualEcsSchema);
+
+export const DraggableValue = pure<{ data: Ecs; fieldName: string }>(({ data, fieldName }) => {
+  const itemDataProvider = {
+    enabled: true,
+    id: escapeDataProviderId(`row-render-value-for-${data._id}-${fieldName}`),
+    name: `${fieldName}: ${getMappedEcsValue({
+      data,
+      fieldName,
+    })}`,
+    queryMatch: {
+      field: getOr(fieldName, fieldName, mappedEcsSchemaFieldNames),
+      value: escapeQueryValue(
+        getMappedEcsValue({
+          data,
+          fieldName,
+        })
+      ),
+    },
+    excluded: false,
+    kqlQuery: '',
+    and: [],
+  };
+
+  const field = allFieldsInSchemaByName[fieldName];
+  const fieldType = field != null ? field.type : '';
+
+  return { data, fieldName } ? (
+    <DraggableWrapper
+      key={`row-render-value-for-${data._id}-${fieldName}`}
+      dataProvider={itemDataProvider}
+      render={(dataProvider, _, snapshot) =>
+        snapshot.isDragging ? (
+          <DragEffects>
+            <Provider dataProvider={dataProvider} />
+          </DragEffects>
+        ) : (
+          <FormattedField data={data} fieldName={fieldName} fieldType={fieldType} />
+        )
+      }
+    />
+  ) : null;
+});
+
+export const SourceIp = pure(({ data }: { data: Ecs }) =>
+  fieldExists({ data, fieldName: 'source.ip' }) ? (
+    <>
+      <EuiFlexItem>
+        <Label>{i18n.SOURCE}</Label>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiFlexGroup gutterSize="none">
+          <EuiFlexItem grow={false}>
+            <DraggableValue
+              data={data}
+              data-test-subj="source-ip-and-port"
+              fieldName={'source.ip'}
+            />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            {fieldExists({ data, fieldName: 'source.port' }) ? ':' : null}
```

Comment:
Given that this function has a `fieldExists` check at the top (to ensure nothing is rendered if `source.ip` does not exist), perhaps we can now remove the ` {fieldExists({ data, fieldName: 'source.port' }) ? ':' : null}` check and instead always show the `:`? 

Put differently, if `source.ip` exists, but `source.port` does not exist, I think this will this render as (for example): `192.168.1.1--`?  If we remove the check, the same scenario would render as `192.168.1.1:--`, with the latter version including the `:` separator between `host` and `port`. Does that make sense? (If yes, a similar check would be removed from the `DestinationIp` component.)

Reply:
It makes sense and I will remove it.